components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -344,14 +345,35 @@ public static NativeObject jsFunction_getApplicationKey(Context cx, Scriptable t
                 String callbackUrl = (String) args[3];
                 String groupingId = (String)args[8];
                 String jsonParams = null;
-                if (args.length == 10) {
+                String applicationId = "";
+                if (args.length >= 10) {
                     jsonParams = (String) args[9];
                 }
 
-                Map<String, Object> keyDetails = getAPIConsumer(thisObj).requestApprovalForApplicationRegistration(
-                        username, applicationName, tokenType, callbackUrl,
-                        accessAllowDomainsArray, validityPeriod, scopes, groupingId,
-                        jsonParams);
+                if (args.length > 10) {
+                    applicationId = (String) args[10];
+                }
+
+                Map<String, Object> keyDetails = null;
+                if (!applicationId.isEmpty()) {
+                    Map<String, Object> appInfoMap = new HashMap<String, Object>();
+                    appInfoMap.put("username", username);
+                    appInfoMap.put("applicationName", applicationName);
+                    appInfoMap.put("tokenType", tokenType);
+                    appInfoMap.put("callbackUrl", callbackUrl);
+                    appInfoMap.put("allowedDomains", accessAllowDomainsArray);
+                    appInfoMap.put("validityPeriod", validityPeriod);
+                    appInfoMap.put("scopes", scopes);
+                    appInfoMap.put("groupingId", groupingId);
+                    appInfoMap.put("jsonParams", jsonParams);
+                    appInfoMap.put("applicationId", applicationId);
+                    keyDetails = getAPIConsumer(thisObj).requestApprovalForApplicationRegistrationByApplicationId(appInfoMap);
+                } else {
+                    keyDetails = getAPIConsumer(thisObj).requestApprovalForApplicationRegistration(
+                            username, applicationName, tokenType, callbackUrl,
```

Comment:
i think we can merge this in to two lines.

Reply:
Fixed with 89b6956b5c1641bd96ee670c827bbf6d0a87550a